hummingbot/market/dolomite/dolomite_market

```diff
@@ -0,0 +1,1362 @@
+        
+        takerGasFees = baseAmount * maxMatches
+        
+        
+
+    async def build_unsigned_order(self, amount: str, price: str, side: str, symbol: str, order_type: OrderType,
+                                   expires: int) -> Dict[str, Any]:
+        
+        
+        url = "%s/v1/orders/prepare" % (self.DOLOMITE_REST_ENDPOINT,)
+        
+        newAccount = web3.eth.account.create() #new auth address / private key 
+        
+        
+        baseToken = symbol.split('-')[0]
+        quoteToken = symbol.split('-')[1]
+        
+        secondaryAmount = 0
+        
+        if order_type is OrderType.MARKET and side == "buy":
+            secondaryAmount = 1000000000000 * math.pow(10, 18)
+            
+        elif order_type is OrderType.MARKET and side == "sell":
+            secondaryAmount = 1e-18 * math.pow(10, 18) #1  
+        else:
+            secondaryAmount = int((float(amount) * float(price)) * math.pow(10, 18))
+            
+            
+            
+        #Set fee amount for takers 
+        
+        fee_url = f"{self.DOLOMITE_REST_ENDPOINT}/v1/info"
+        
+        
+        res = await self._api_request(http_method="get", url=fee_url)
+        
+        
+        #Gas fees 
+        
+        feeData = res["data"]["base_spot_trading_fee_amounts"] #find by quoteToken
+        
+        baseAmount = float(feeData[quoteToken]["amount"]) / math.pow(10, 18)
+        maxMatches = res["data"]["max_number_of_taker_matches_per_order"]
+        
+        takerGasFees = baseAmount * maxMatches
+        
+        
+        #Commission fees
+        
+        market_url = f"{self.DOLOMITE_REST_ENDPOINT}/v1/markets/{symbol}"
+        
+        res2 = await self._api_request(http_method="get", url=market_url)
+        
+        marketPrice = float(res2["data"]["current_price"]["amount"]) / math.pow(10, 18)
+        
+        takerRate = float(res["data"]["taker_fee_percentage"])
+        
+        commissionAmount = marketPrice * takerRate * float(amount)
+        
+        #Total 
+        totalTakerFee = takerGasFees + commissionAmount #max amount - set upperbound?
+
+        
+        #fee address
+        feeAddress = res["data"]["fee_collecting_wallet_address"]
+        
+        data = {
+            
+            "fee_collecting_wallet_address": feeAddress, 
+            "wallet_split_percentage": 0,
+            "owner_address": Web3Wallet.address,
+            "auth_address": str(newAccount.address),
+            "order_side": side.upper(),
+            "order_type": "MARKET" if order_type is OrderType.MARKET else "LIMIT",
+            "auth_private_key": str(binascii.hexlify(newAccount.privateKey).decode("utf-8")),
+            "market": tickers[baseToken] + "-" + tickers[quoteToken],
+            "primary_padded_amount": str(int(float(amount) * math.pow(10, 18))),
+            "secondary_padded_amount": str(secondaryAmount),
+            "creation_timestamp": int(time.time()) * 1000,
```

Comment:
For simplicity's sake, subtract 5 seconds here. Reason why is complex: TLDR latency issues with the Ethereum network